pkg/kvstore/allocator/allocator.go

```diff
@@ -380,21 +383,32 @@ var (
 	idRandomizerMutex lock.Mutex
 )
 
-// Naive ID allocation mechanism.
-func (a *Allocator) selectAvailableID() (ID, string) {
+func (a *Allocator) newRandomIDs() {
 	idRandomizerMutex.Lock()
-	defer idRandomizerMutex.Unlock()
+	a.randomIDs = idRandomizer.Perm(int(a.max - a.min + 1))
+	idRandomizerMutex.Unlock()
+}
 
+// Naive ID allocation mechanism.
+func (a *Allocator) selectAvailableID() (ID, string) {
 	a.mutex.RLock()
 	defer a.mutex.RUnlock()
 
-	tried := 0
+	for attempt := 0; attempt < 2; attempt++ {
+		for i, r := range a.randomIDs {
+			id := ID(r) + a.min
+			if _, ok := a.cache[id]; !ok && a.localKeys.lookupID(id) == "" {
+				// remove the previously tried IDs that are already in
+				// use from the list of IDs to attempt allocation
+				if len(a.randomIDs) > 0 {
```

Comment:
Remove this test. This is always true since we're currently iterating over that slice.